packages/contracts/contracts/LenderCommitmentForwarder/extensions/LenderCommitmentGroup/LenderCommitmentGroup_Smart.sol

```diff
@@ -736,22 +747,77 @@ contract LenderCommitmentGroup_Smart is
     function abs(int x) private pure returns (uint) {
         return x >= 0 ? uint(x) : uint(-x);
     }
- 
-    function getCollateralRequiredForPrincipalAmount(uint256 _principalAmount)
-        public
+
+
+    function calculateCollateralRequiredToBorrowPrincipal(  
+        uint256 _principalAmount
+    ) public
         view
-        returns (uint256)
-    {
-        uint256 baseAmount = _calculateCollateralTokensAmountEquivalentToPrincipalTokens(
+        virtual
+        returns (uint256) {
+
+        uint256 baseAmount = calculateCollateralTokensAmountEquivalentToPrincipalTokens(
                 _principalAmount
-            );
+        ); 
 
         //this is an amount of collateral
         return baseAmount.percent(collateralRatio);
     }
 
+
+    //this is expanded by 10e18
+    //this logic is very similar to that used in LCFA 
+    function calculateCollateralTokensAmountEquivalentToPrincipalTokens(
+        uint256 principalAmount 
+    ) public view virtual returns (uint256 collateralTokensAmountToMatchValue) {
+        //same concept as zeroforone
+       // (address token0, ) = _getPoolTokens();
+
+        //bool principalTokenIsToken0 = (address(principalToken) == token0);
+         //uint256 pairPriceImmediate = _getUniswapV3TokenPairPrice(0);
+
+        uint256 pairPriceWithTwapFromOracle = UniswapPricingLibrary
+            .getUniswapPriceRatioForPoolRoutes(poolOracleRoutes);
+       
+      
+       //uint256 scaledPoolOraclePrice = UniswapPricingLibrary.getUniswapPriceRatioForPoolRoutes(
+       //         poolOracleRoutes
+       //     ).percent(collateralRatio);
+ 
+
+        uint256 maxPrincipalPerCollateralAmount = maxPrincipalPerCollateralAmount == 0  
+                ? pairPriceWithTwapFromOracle   
+                : Math.min(
+                    pairPriceWithTwapFromOracle,
```

Comment:
Isn't this only the current price? ie it's not taking into the collateral ratio %. so if the price is used, it should be multiplying by the collateral ratio %